src/components/react-admin/DeleteWithConfirmButtonJS.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import { fade } from '@material-ui/core/styles/colorManipulator';
+import ActionDelete from '@material-ui/icons/Delete';
+import { Button,
+  Confirm,
+  useDeleteWithConfirmController,
+  useResourceContext,
+} from 'react-admin';
+
+import classnames from 'classnames';
+
+const defaultIcon = <ActionDelete />;
+
+const useStyles = makeStyles(
+  theme => ({
+    deleteButton: {
+      color: theme.palette.error.main,
+      '&:hover': {
+        backgroundColor: fade(theme.palette.error.main, 0.12),
+        // Reset on mouse devices
+        '@media (hover: none)': {
+          backgroundColor: 'transparent',
+        },
+      },
+    },
+  }),
+  { name: 'RaDeleteWithConfirmButtonJS' },
+);
+
+const DeleteWithConfirmButtonJS = props => {
+  const {
+    basePath,
+    classes: classesOverride,
+    className,
+    confirmTitle = 'ra.message.delete_title',
+    confirmContent = 'ra.message.delete_content',
+    icon = defaultIcon,
+    label = 'ra.action.delete',
+    mutationMode,
+    onClick,
+    record,
+    redirect = 'list',
+    ressource,
+    onSuccess,
+    onFailure,
+    idFields,
```

Comment:
What if we want to use 2 fields in our message? Not important for our needs, but for a PR on React-admin, should be more reusable to have something like `translateOptions` direct overiding? What do you think??

Reply:
I'll try passing `translateOptions` as props instead of IdFields